src/storage/engine/perf_context.rs

```diff
@@ -0,0 +1,69 @@
+// Copyright 2018 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use rocksdb::PerfContext;
+
+macro_rules! define_perf_statistics {
```

Comment:
I am confused that why using macro here, why using two classes (Instant and Delta) here.

Reply:
macro is used to reduce field duplications, for example, in delta(), add(), new(), every fields are just repeated. by using macros, we only need to define field once, which will be very easy to maintain.

two classes are naturally introduced because they have totally different meanings, just like `Instant` and `Duration` in `std::time`. The only common part of the two classes is that they have same fields. As you can see, these two classes in fact does not even have a same method.